assets/src/edit-story/elements/text/output.js

```diff
@@ -14,63 +14,75 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
 /**
  * Internal dependencies
  */
 import StoryPropTypes from '../../types';
 import generatePatternStyles from '../../utils/generatePatternStyles';
 import { dataToEditorX, dataToEditorY } from '../../units';
-import { draftMarkupToContent, generateFontFamily } from './util';
+import { draftMarkupToContent, generateParagraphTextStyle } from './util';
 
 /**
- * Returns AMP HTML for saving into post content for displaying in the FE.
+ * Renders DOM for the text output based on the provided unit converters.
  */
-function TextOutput({
-  element: {
-    bold,
-    content,
-    color,
-    backgroundColor,
-    fontFamily,
-    fontFallback,
-    fontSize,
-    fontWeight,
-    fontStyle,
-    letterSpacing,
-    lineHeight,
-    padding,
-    textAlign,
-    textDecoration,
-  },
-  box: { width },
+export function TextOutputWithUnits({
+  element: { bold, content, color, backgroundColor, padding, ...rest },
+  dataToStyleX,
+  dataToStyleY,
+  paddingToStyle,
+  className,
 }) {
-  const horizontalPadding = dataToEditorX(padding.horizontal, width);
-  // The padding % is taken based on width, thus using X and width for vertical, too.
-  const verticalPadding = dataToEditorX(padding.vertical, width);
+  const paddingStyles = paddingToStyle(padding);
   const style = {
-    fontSize: `${dataToEditorY(fontSize, 100)}%`,
-    fontStyle: fontStyle ? fontStyle : null,
-    fontFamily: generateFontFamily(fontFamily, fontFallback),
-    fontWeight: fontWeight ? fontWeight : null,
-    lineHeight,
-    letterSpacing: letterSpacing ? letterSpacing + 'em' : '0',
-    padding: padding ? `${verticalPadding}% ${horizontalPadding}%` : '0',
-    textAlign: textAlign ? textAlign : null,
-    textDecoration,
-    whiteSpace: 'pre-wrap',
+    ...generateParagraphTextStyle(rest, dataToStyleX, dataToStyleY),
     ...generatePatternStyles(backgroundColor),
     ...generatePatternStyles(color, 'color'),
+    padding: `${paddingStyles.vertical} ${paddingStyles.horizontal}`,
```

Comment:
Then this one isn't necessary anymore, right?

Reply:
Ah, it is necessary because `paddingStyles` aren't sent to `generateParagraphTextStyle` as `padding`. Should it?

Reply:
So, yeah, output is a little different. We can't express paddings proportionally in the actual CSS. CSS paddings `%` are defined in the basis of the element's width: both vertical and horizontal. So, there's a special override here. But I think it's ok for now. Once you have highlight, we will need to see how much more that would change.